source/common/router/config_utility.h

```diff
@@ -25,14 +27,49 @@ namespace Router {
  */
 class ConfigUtility {
 public:
+  enum class HeaderMatchType { Value, Regex, Range };
+
   struct HeaderData {
+    // HeaderMatcher will consist of one of the below two options :
+    // 1) value (string) and regex (bool)
     // An empty header value allows for matching to be only based on header presence.
     // Regex is an opt-in. Unless explicitly mentioned, the header values will be used for
     // exact string matching.
-    HeaderData(const envoy::api::v2::route::HeaderMatcher& config)
-        : name_(config.name()), value_(config.value()),
-          is_regex_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, regex, false)),
-          regex_pattern_(is_regex_ ? RegexUtil::parseRegex(value_) : std::regex()) {}
+    // This is now deprecated.
+    // 2) header_match_specifier which can be any one of exact_match, regex_match or range_match.
+    // Absence of these options implies empty header value  => match based on header presence.
+    // exact_match value will be used for exact string matching.
+    // regex_match : Match will succeed if header value matches the value specified in regex_match.
+    // range_match : Match will succeed if header value lies within the range specified in this
+    // field, using half open interval semantics [start,end)
```

Comment:
nits: remove "=>" and extra spaces; remove space before colons; period at the end of the last sentence.

I think it would be easier to read this comment if you indented the lines below the two options.